pkg/filesystem/nydus/resolver.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2021. Alibaba Cloud. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package nydus
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/containerd/containerd/reference/docker"
+	"github.com/containerd/nydus-snapshotter/pkg/auth"
+	"github.com/containerd/nydus-snapshotter/pkg/utils/registry"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/pkg/errors"
+)
+
+type Resolver struct {
+	transport http.RoundTripper
+}
+
+func NewResolver() *Resolver {
+	resolver := Resolver{
+		transport: http.DefaultTransport,
+	}
+	return &resolver
+}
+
+func (r *Resolver) Resolve(ref, digest string, labels map[string]string) (io.ReadCloser, error) {
+	named, err := docker.ParseDockerRef(ref)
+	if err != nil {
+		return nil, err
```

Comment:
We'd better use `errors.Wrap` wrap all `err` as far as possible.

Reply:
fixed